### ISSUE TITLE
can you share a working circuit for VFD along with arduino?

### ISSUE THREAD

**Issue by amrithmmh** [carneeki/OpenVFD#1]
i am really interested in your project thanks

**Comment by carneeki** [comment]
This project had to take a bit of a back seat for a while, but I want to get back on it soon!

I've just pushed some cad files I was working on a while ago, I'm not sure if they will work, but feel free to review them. If nothing else, the block diagram might be a good place to start a new VFD project.

**Comment by amrithmmh** [comment]
you might be interested in this
http://www.microchip.com//wwwAppNotes/AppNotes.aspx?appnote=en567897
http://ww1.microchip.com/downloads/en/AppNotes/00001660B.pdf

I would like to help you with this project but I dont have much knowledge or experience

**Comment by carneeki** [comment]
These application notes are great thanks! It makes me the idea in my head might be on the right track.

As for knowledge or experience, that hasn't stopped me in the past! My strategy is to set myself some projects and work out what I need to learn, and then learn it. With mixed results ;)

**Comment by carneeki** [comment]
As I don't know what your experience is... I should mention a word of caution though: VFDs typically operate at voltages that can be lethal. If you're new to electronics, I would recommend starting off with some low voltage projects (anything between 0 and 12V should be fine) to build up your skills, and use bought power supplies that have passed the important safety checks.

**Comment by amrithmmh** [comment]
there are many ICs from microchip too that have application notes you can check out those too
and one more IC
https://www.nxp.com/docs/en/data-sheet/MC3PHAC.pdf
http://application-notes.digchip.com/314/314-67275.pdf
http://sroogle.ru/?q=MC3PHAC
http://www.electronics-lab.com/project/3-phase-ac-motor-controller/
MC3PHAC is retired though :(

**Comment by carneeki** [comment]
True, I've got a couple of PIC chips in a drawer somewhere, however, I'm thinking something like the Arduino Nano will have less of a steep learning curve to people like CNC machinists who may be wanting to build a VFD. Plug USB cable in and push the 'Upload' button.

If something like a USBasp or PICkit are needed for ICSP, that adds an additional barrier to entry. If I were making a commercial product, I'd look at going either PIC or AVR without the Arduino framework, but keeping it accessible for the mechatronics enthusiast is important.

**Comment by dexterbot80** [comment]
![openvfd](https://user-images.githubusercontent.com/47325794/52440651-65042480-2b27-11e9-91bf-718edecd5aa1.JPG)
Hello, Here is an update for testing PCB . I hope it will work :)

**Comment by dexterbot80** [comment]
What are the TRIG and / EN pins? Thank you

**Comment by carneeki** [comment]
Hi @dexterbot80  the board looks great!

TRIG is to trigger an oscilloscope (during testing / calibration) - see line `19` in `OpenVFD.ino` ;)
/EN I think was to be an "Enable" pin where pulling it low would enable the PWM outputs and pulling it high would disable the outputs, this is because controllers like Grbl often use a "Spindle Enable" output pin to tell the VFD to start the spindle, but it could also be used for a manual switch. A simple `if()` inside of `loop()` should be all that's needed to make it functional.

**Comment by dexterbot80** [comment]
![openvfd](https://user-images.githubusercontent.com/47325794/52473081-9e777700-2b9d-11e9-9d93-f8bb9f74e39b.JPG)


Thank you
